FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/mqtt-agent-task.c

```diff
@@ -0,0 +1,955 @@
+ */
+static BaseType_t prvSocketConnect( NetworkContext_t * pxNetworkContext );
+
+/**
+ * @brief Disconnect a TCP connection.
+ *
+ * @param[in] pxNetworkContext Network context.
+ *
+ * @return `pdPASS` if disconnect succeeds, else `pdFAIL`.
+ */
+static BaseType_t prvSocketDisconnect( NetworkContext_t * pxNetworkContext );
+
+/**
+ * @brief Callback executed when there is activity on the TCP socket that is
+ * connected to the MQTT broker.  If there are no messages in the MQTT agent's
+ * command queue then the callback send a message to ensure the MQTT agent
+ * task unblocks and can therefore process whatever is necessary on the socket
+ * (if anything) as quickly as possible.
+ *
+ * @param[in] pxSocket Socket with data, unused.
+ */
+static void prvMQTTClientSocketWakeupCallback( Socket_t pxSocket );
+
+/**
+ * @brief Fan out the incoming publishes to the callbacks registered by different
+ * tasks. If there are no callbacks registered for the incoming publish, it will be
+ * passed to the unsolicited publish handler.
+ *
+ * @param[in] pMqttAgentContext Agent context.
+ * @param[in] packetId Packet ID of publish.
+ * @param[in] pxPublishInfo Info of incoming publish.
+ */
+static void prvIncomingPublishCallback( MQTTAgentContext_t * pMqttAgentContext,
+                                        uint16_t packetId,
+                                        MQTTPublishInfo_t * pxPublishInfo );
+
+/**
+ * @brief Function to attempt to resubscribe to the topics already present in the
+ * subscription list.
+ *
+ * This function will be invoked when this demo requests the broker to
+ * reestablish the session and the broker cannot do so. This function will
+ * enqueue commands to the MQTT Agent queue and will be processed once the
+ * command loop starts.
+ *
+ * @return `MQTTSuccess` if adding subscribes to the command queue succeeds, else
+ * appropriate error code from MQTTAgent_Subscribe.
+ * */
+static MQTTStatus_t prvHandleResubscribe( void );
+
+/**
+ * @brief Passed into MQTTAgent_Subscribe() as the callback to execute when the
+ * broker ACKs the SUBSCRIBE message. This callback implementation is used for
+ * handling the completion of resubscribes. Any topic filter failed to resubscribe
+ * will be removed from the subscription list.
+ *
+ * See https://freertos.org/mqtt/mqtt-agent-demo.html#example_mqtt_api_call
+ *
+ * @param[in] pxCommandContext Context of the initial command.
+ * @param[in] pxReturnInfo The result of the command.
+ */
+static void prvSubscriptionCommandCallback( void * pxCommandContext,
+                                            MQTTAgentReturnInfo_t * pxReturnInfo );
+
+/**
+ * @brief Task used to run the MQTT agent.  In this example the first task that
+ * is created is responsible for creating all the other demo tasks.  Then,
+ * rather than create prvMQTTAgentTask() as a separate task, it simply calls
+ * prvMQTTAgentTask() to become the agent task itself.
```

Comment:
Instead of explaining about not creating a separate task with this function, it may be clearer to just call it something like `prvRunMqttAgent()` as the task is represented by the `prvConnectAndCreateDemoTasks` function..